testutil/ciao-down/cloudinit.go

```diff
@@ -232,3 +232,146 @@ const metaDataTemplate = `
+ - echo "PATH=$PATH:/usr/local/go/bin:{{$.GoPath}}/bin"  >> /etc/environment
+ - {{template "PROXIES" .}}wget https://storage.googleapis.com/golang/go1.7.4.linux-amd64.tar.gz -O /tmp/go1.7.4.linux-amd64.tar.gz
+ - {{template "CHECK" .}}
+ - curl -X PUT -d "Unpacking Go" 10.0.2.2:{{.HTTPServerPort}}
+ - tar -C /usr/local -xzf /tmp/go1.7.4.linux-amd64.tar.gz
+ - {{template "CHECK" .}}
+ - rm /tmp/go1.7.4.linux-amd64.tar.gz
+
+ - groupadd docker
+ - sudo gpasswd -a {{.User}} docker
+ - curl -X PUT -d "Installing apt-transport-https and ca-certificates" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get install apt-transport-https ca-certificates
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Add docker GPG key" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Retrieving updated list of packages" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get update
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Upgrading" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get upgrade -y
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Installing Docker" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get install docker-engine -y
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Installing GCC" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get install gcc -y
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Installing Make" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get install make -y
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Installing QEMU" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get install qemu-system-x86 -y
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Installing xorriso" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get install xorriso -y
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Auto removing unused components" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get auto-remove -y
+ - {{template "CHECK" .}}
+
+ - curl -X PUT -d "Installing Go development utils" 10.0.2.2:{{.HTTPServerPort}}
+ - sudo -u {{.User}} {{template "PROXIES" .}} GOPATH={{.GoPath}} /usr/local/go/bin/go get github.com/fzipp/gocyclo github.com/gordonklaus/ineffassign github.com/golang/lint/golint github.com/client9/misspell/cmd/misspell
+ - {{template "CHECK" .}}
+
+ - chown {{.User}}:{{.User}} -R {{.GoPath}}
+
+{{if len .GitUserName}}
+ - curl -X PUT -d "Setting git user.name" 10.0.2.2:{{.HTTPServerPort}}
+ - sudo -u {{.User}} git config --global user.name "{{.GitUserName}}"
+ - {{template "CHECK" .}}
+{{end}}
+{{if len .GitEmail}}
+ - curl -X PUT -d "Setting git user.email" 10.0.2.2:{{.HTTPServerPort}}
+ - sudo -u {{.User}} git config --global user.email {{.GitEmail}}
+ - {{template "CHECK" .}}
+{{end}}
+ - curl -X PUT -d "FINISHED" 10.0.2.2:{{.HTTPServerPort}}
```

Comment:
In the ciao version of this file we have a %s here, which is used by the --runcmd option to insert extra user supplied commands.  I use this to install emacs and set up my git aliases, etc.